platform/newsfeed/src/main.rs

```diff
@@ -0,0 +1,212 @@
+                        warn!("Error while deserializing message payload: {:?}", e);
+                        ""
+                    }
+                };
+
+                let message: NewsFeedSchema = match serde_json::from_str(&payload) {
+                    Ok(result) => result,
+                    Err(e) => {
+                        warn!("Failed to deserialize message: {:?}", e);
+                        continue;
+                    }
+                };
+
+                let news_source_url: String = match Url::parse(&message.fields.1.doc) {
+                    Ok(result) => {
+                        let news_source_url = result.as_str().to_string();
+                        info!("Received news source: {}", news_source_url);
+                        news_source_url
+                    }
+                    Err(e) => {
+                        warn!("Failed to parse news source url: {:?}", e);
+                        continue;
+                    }
+                };
+
+                let pattern = match Regex::new(r"\s+") {
+                    Ok(result) => result,
+                    Err(e) => {
+                        warn!("Failed to compile regex: {:?}", e);
+                        continue;
+                    }
+                };
+
+                let raw: String = match reqwest::get(news_source_url).await {
+                    Ok(response) => match response.text().await {
+                        Ok(result) => result,
+                        Err(e) => {
+                            error!("Failed to fetch news source text response: {:?}", e);
+                            continue;
+                        }
+                    },
+                    Err(e) => {
+                        error!("Failed to fetch news source: {:?}", e);
+                        continue;
+                    }
+                };
+
+                let body_selector = Selector::parse("body").unwrap();
+                let document = Html::parse_document(&raw);
+                let body = document
+                    .select(&body_selector)
+                    .next()
+                    .map(|element| element.text().collect::<Vec<_>>().join(" "))
+                    .map(|body| pattern.replace_all(&body, " ").into_owned())
+                    .unwrap();
+
+                let chat = ClaudeMessage {
+                    model: ClaudeModel::Claude35Sonnet,
+                    max_tokens: 1024,
+                    system: Some("You are a financial news analyst, and you are extracting the content from an html source with noise".to_string()),
+                    messages: vec![
+                        Prompt {
+                            role: Role::User,
+                            content: body,
+                        },
+                    ],
+                };
+
+                let chat_response = chat.generate().await;
+
+                info!("chat response: {:?}", chat_response);
```

Comment:
Same comment here as above regarding logs.

Reply:
switched to `debug!`